frontend/src/components/RequireClaimsButtons.vue

```diff
@@ -1,48 +1,79 @@
 <template>
   <span>
-    <v-btn :key="button.text" v-for="button in buttons" v-bind="button.props">
+    <v-btn :key="`${button.text}-${index}`" v-for="(button, index) in routeButtons" v-bind="button.props">
       {{ button.text }}
     </v-btn>
   </span>
 </template>
 
 <script lang="ts">
 import { Component, Vue } from "vue-property-decorator";
-import firebase from "firebase/app";
-import "firebase/auth";
+import _ from "lodash";
+import { mapActions } from "vuex";
+import { RouteConfig } from "vue-router";
+import { filterRoutesByClaims, hasClaim } from "@/router";
 
 @Component({
-  name: "RequireClaimsButtons"
+  name: "RequireClaimsButtons",
+  methods: {
+    ...mapActions("user", ["getUserClaims"])
+  }
 })
 export default class RequireClaimsButtons extends Vue {
   userClaims: any = {};
-  initialButtons = [
-    {
-      claims: { TE: true },
-      props: {
-        to: "te/upload",
-        color: "primary"
-      },
-      text: "Upload"
-    }
-  ];
+  routeButtons: any[] = [];
 
-  mounted() {
-    this.getUserClaims();
+  async mounted() {
+    this.userClaims = (await this.getUserClaims()) || {};
+    this.makeButtons();
   }
 
-  get buttons() {
-    return this.initialButtons.filter((button: any) => {
-      return Object.entries(this.userClaims).some(([claimName, claimValue]) => {
-        return button.claims[claimName] === claimValue;
-      });
-    });
-  }
+  makeButtons() {
```

Comment:
Personally, I'd move this filtering logic into the router itself. Why not to export something like `getHomePageRoutes` from the router and just transform the routes into the buttons here in this component? I think it will satisfy SoC principle better.